test/init/initUtils.test.js

```diff
@@ -0,0 +1,223 @@
+const fs = require('fs');
+const initPackageJson = require('init-package-json');
+const { asyncPackageInit, buildWeb3Wallet, validateEmptyDir } = require('../../dist/src/cmds/init/initUtils');
+const readlineUtils = require('../../dist/src/utils/readline');
+const miscUtils = require('../../dist/src/utils/misc');
+
+jest.mock('fs', () => ({
+  existsSync: jest.fn(),
+  promises: {
+    mkdir: jest.fn(),
+    readdir: jest.fn(),
+    readFile: jest.fn(),
+  },
+}));
+
+jest.mock('init-package-json');
+
+describe('initUtils', () => {
+
+  describe('asyncPackageInit', () => {
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.restoreAllMocks();
+    });
+
+    it('console logs if successful', async () => {
+      const existsSyncMock = jest.spyOn(fs, 'existsSync').mockImplementation(() => true);
+      const readFileMock = jest.spyOn(fs.promises, 'readFile').mockImplementationOnce();
+      const parseMock = jest.spyOn(JSON, 'parse').mockImplementation();
+      jest.spyOn(console, 'log').mockImplementation();
+      await asyncPackageInit();
+      expect(existsSyncMock).toHaveBeenCalledTimes(1);
+      expect(readFileMock).toHaveBeenCalledTimes(1);
+      expect(parseMock).toHaveBeenCalledTimes(1);
+      expect(global.console.log).toHaveBeenCalledTimes(2);
+    });
+
+    it('throws error in catch block', async () => {
```

Comment:
This test name is explicitly referencing an implementation detail. We should try to reframe it in terms of its behavior as observed from the outside.